### ISSUE TITLE
Original exception never gets logged in Retry Topic flow

### ISSUE THREAD

**Issue by tomazfernandes** [spring-projects/spring-kafka#2212]
This [SO](https://stackoverflow.com/questions/71719065/backoffexceptions-are-logged-at-error-level-when-using-retrytopicconfiguration) user pointed out that, when the record goes through the Retry Topic flow, the original exception never gets logged.

We had already discussed this a bit earlier this year. 

@garyrussell [proposed](https://github.com/spring-projects/spring-kafka/issues/2063#issuecomment-1010404269_):

> But I see your point; since we always recover these failures, they are never logged. Maybe the DLPR needs an option to log the exception after successfully recovering a record (but that doesn't need to be part of this fix).

I suggested as an [alternative](https://github.com/spring-projects/spring-kafka/issues/2063#issuecomment-1010482981_):

> We could log the error message just before we either send the record to the DLT or give up processing - that way nothing gets logged if any of the retrials work, and gets logged only once if none.

The main difference seems to be whether we want to log the exception at every retry attempt or only before it goes to the DLT.

We might look into adding both and making these configurable - users might be able to set a flag in `DLPR`  to log after every attempt, and `DLPRF` to log before going to the DLT.

We could also have this as an enum in `DLPRF`, with something like `OriginalExceptionLoggingStrategy.NEVER`, `EVERY_ATTEMPT`, `AFTER_RETRIES_EXHAUSTED` - `DLPRF` would then set the appropriate flag in `DLPR`.

WDYT?

_Originally posted by @tomazfernandes in https://github.com/spring-projects/spring-kafka/discussions/2211_

**Comment by tomazfernandes** [comment]
Looks like `Reference this discussion in new issue` is not the same as `converting discussion into issue` as you do 😄

Guess it's alright though - let me know if otherwise. I should open a PR for this shortly if that's ok.

Thanks.

**Comment by garyrussell** [comment]
It doesn't really make a difference; this is fine.
